source/request_response.c

```diff
@@ -573,6 +573,21 @@ int aws_http1_stream_write_chunk(struct aws_http_stream *http1_stream, const str
     return http1_stream->vtable->http1_write_chunk(http1_stream, options);
 }
 
+int aws_h2_stream_write_data(struct aws_http_stream *h2_stream, const struct aws_h2_data_options *options) {
+    AWS_PRECONDITION(h2_stream);
+    AWS_PRECONDITION(h2_stream->vtable);
+    AWS_PRECONDITION(options);
+    if (!h2_stream->vtable->http2_write_data) {
+        AWS_LOGF_TRACE(
+            AWS_LS_HTTP_STREAM,
+            "id=%p: H2 stream only function (write_data) invoked on other stream, ignoring call",
+            (void *)h2_stream);
+        return aws_raise_error(AWS_ERROR_INVALID_STATE);
+    }
```

Comment:
there is an old [pr](https://github.com/awslabs/aws-c-http/pull/327/files#diff-d7eec29a8674c2301bf0ced41b2720d9f20faa547a45348d4140915aa9dd323aR346), we want to enforce (otherwise crash) only h2 connection/stream used at those version specific functions. Mike convinced me that user cannot do anything to recover from this kind of error. We'd better just crash. So, maybe instead of handling it, just `AWS_PRECONDITION()`

Reply:
Oh, sweet. I was just copying the local code, but I'll do that instead.